### PR TITLE
Refactor testnet SPO certificates functions

### PR DESCRIPTION
**Pull request by carbolymer**
# Description

Uses new CLI format and supports Conway era. Remove redundant usage of `CPP` extensions.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs
Comment (carbolymer) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs:1]: [ ] Remove CPP

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/InfoAction.hs
Comment (carbolymer) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/InfoAction.hs:1]: * [ ] Remove CPP

File: cardano-testnet/src/Testnet/Property/Utils.hs [hunk 1]
```diff
@@ -122,3 +127,18 @@ runInBackground act = void . H.evalM $ allocate (H.async act) cleanUp
 
 decodeEraUTxO :: (IsShelleyBasedEra era, MonadTest m) => ShelleyBasedEra era -> Aeson.Value -> m (UTxO era)
 decodeEraUTxO _ = H.jsonErrorFail . Aeson.fromJSON
+
+
+requireEon :: forall eon era m. Eon eon
```
Comment (smelc) [inline cardano-testnet/src/Testnet/Property/Utils.hs:132]: I think an haddock explaining what this function does would be beneficial. The type doesn't make obvious what's the actual era compared to the expected era.

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs [hunk 2]
```diff
@@ -1,5 +1,4 @@
 {-# LANGUAGE BlockArguments #-}
-{-# LANGUAGE CPP #-}
```
Comment (smelc) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs:2]: Is this change intended?

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs [hunk 3]
```diff
@@ -7,12 +7,6 @@
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TypeApplications #-}
 
-#if __GLASGOW_HASKELL__ >= 908
-{-# OPTIONS_GHC -Wno-x-partial #-}
-#endif
```
Comment (smelc) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs:12]: Is this change intended?

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs [hunk 4]
```diff
@@ -64,7 +58,7 @@ hprop_drep_retirement = H.integrationRetryWorkspace 2 "drep-retirement" $ \tempA
   TestnetRuntime
     { testnetMagic
     , poolNodes
-    , wallets
+    , wallets=wallet0:_
```
Comment (smelc) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs:61]: Haha nice trick to avoid the `!!`/`head` below!

**Comment by smelc** [comment]
Note: it is probably conflicting with https://github.com/IntersectMBO/cardano-node/pull/5732

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs (hunk 2 above)
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs:2]: Yes, we don't need CPP here anymore

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs (hunk 3 above)
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs:12]: Yes, warning is no longer raised here.

File: cardano-testnet/src/Testnet/Property/Utils.hs (hunk 1 above)
Comment (carbolymer) [reply cardano-testnet/src/Testnet/Property/Utils.hs:134]: added

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs (hunk 2 above)
Comment (smelc) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs:2]: Nice :+1:

**Review by smelc — APPROVED** [review]

File: cardano-testnet/src/Testnet/Property/Utils.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-testnet/src/Testnet/Property/Utils.hs:134]: Where do you use this function?

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs [hunk 5]
```diff
@@ -64,7 +57,7 @@ hprop_drep_retirement = H.integrationRetryWorkspace 2 "drep-retirement" $ \tempA
...
```
Comment (newhoggy) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs:60]: This can still fail right?  What happens when it does?  Do you get nice output?

File: cardano-testnet/src/Testnet/Property/Utils.hs (hunk 1 above)
Comment (carbolymer) [reply cardano-testnet/src/Testnet/Property/Utils.hs:134]: Actually, I'm not using it anymore - I've refactored with `monoidForEraInEon`. I'll remove dead code.

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs (hunk 5 above)
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs:60]: It just prints:
```
            Pattern match failure in 'do' block at test/cardano-testnet-test/Cardano/Testnet/Test/SubmitApi/Babbage/Transaction.hs:(68,3)-(73,5)
```
This is a testnet setup so it's pretty easy to spot on when working on a test.

Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs:60]: Ideally we should use a dependent list here.
